packages/eas-cli/src/commands/release/publish.ts

```diff
@@ -0,0 +1,152 @@
+    let {
+      flags: { json: jsonFlag, releaseName, runtimeVersion, publishMessage, inputDir },
+    } = this.parse(ReleasePublish);
+
+    const projectDir = await findProjectRootAsync(process.cwd());
+    if (!projectDir) {
+      throw new Error('Please run this command inside a project directory.');
+    }
+    const accountName = await getProjectAccountNameAsync(projectDir);
+    const {
+      exp: { slug },
+    } = getConfig(projectDir, { skipSDKVersionRequirement: true });
+    const projectId = await ensureProjectExistsAsync({
+      accountName,
+      projectName: slug,
+    });
+
+    if (!inputDir) {
+      const validationMessage = 'input directory must be specified.';
+      if (jsonFlag) {
+        throw new Error(validationMessage);
+      }
+      ({ inputDir } = await promptAsync({
+        type: 'text',
+        name: 'inputDir',
+        message: 'Please enter the name of the directory created by the bundler:',
+        initial: 'dist',
+        validate: value => (value ? true : validationMessage),
+      }));
+    }
+
+    if (!releaseName) {
+      const validationMessage = 'release name may not be empty.';
+      if (jsonFlag) {
+        throw new Error(validationMessage);
+      }
+      ({ releaseName } = await promptAsync({
+        type: 'text',
+        name: 'releaseName',
+        message: 'Please enter the name of the release to publish on:',
+        validate: value => (value ? true : validationMessage),
+      }));
+    }
+
+    if (!runtimeVersion) {
+      const validationMessage = 'runtimeVersion name may not be empty.';
+      if (jsonFlag) {
+        throw new Error(validationMessage);
+      }
+      ({ runtimeVersion } = await promptAsync({
+        type: 'text',
+        name: 'runtimeVersion',
+        message: 'Please enter the runtime version of the updates',
+        validate: value => (value ? true : validationMessage),
+      }));
+    }
+
+    if (!publishMessage) {
+      const validationMessage = 'publish message may not be empty.';
+      if (jsonFlag) {
+        throw new Error(validationMessage);
+      }
+      ({ publishMessage } = await promptAsync({
+        type: 'text',
+        name: 'publishMessage',
+        message: `Please enter a publication message.`,
+        initial: (await getLastCommitMessageAsync())?.trim(),
+        validate: value => (value ? true : validationMessage),
+      }));
+    }
+
+    const { id: releaseId } = await getReleaseByNameAsync({
+      appId: projectId,
+      releaseName: releaseName!,
+    });
+
+    log.withTick(
+      `️Publishing "${publishMessage}" to ${releaseName}@${runtimeVersion} to on project ${chalk.bold(
+        `@${accountName}/${slug}`
+      )}.`
+    );
+
+    log.withTick('Collecting assets...');
+    const assets = collectAssets(inputDir!);
+
+    log.withTick('Uploading assets...');
+    await uploadAssetsAsync(assets);
+    const updateInfoGroup = buildUpdateInfoGroup(assets);
+
+    log.withTick('Publishing...');
```

Comment:
If `--json` is specified, these logs break the output, which then can't be parsed as JSON.

I think the correct way to do this would be to only print the final JSON output to `stdout` and print these status messages to `stderr` (idea from https://clig.dev/#the-basics).

Reply:
I removed all of the log.ticks and replaced them with `ora` spinners or only output if the json flag is not set.